internal/resources/fetching/fetchers/aws/iam_fetcher.go

```diff
@@ -131,7 +131,22 @@ func (r IAMResource) GetMetadata() (fetching.ResourceMetadata, error) {
 }
 
 func (r IAMResource) GetElasticCommonData() (map[string]any, error) {
-	return map[string]any{
+	m := map[string]any{
 		"cloud.service.name": "IAM",
-	}, nil
+	}
+
+	switch r.AwsResource.GetResourceType() {
```

Comment:
Just wondering, I see some switch casing around on this. Maybe is it an idea to have two separate structs, one for IAMUserType and one for PolicyType?

Reply:
If you think it's not worth it, I trust in your call.

I just find such switch cases easy to miss in refactors and search